tokio/src/runtime/metrics/batch.rs

```diff
@@ -39,87 +52,107 @@ pub(crate) struct MetricsBatch {
-        worker.mean_poll_time.store(mean_poll_time, Relaxed);
-        worker.park_count.store(self.park_count, Relaxed);
-        worker
-            .park_unpark_count
-            .store(self.park_unpark_count, Relaxed);
-        worker.noop_count.store(self.noop_count, Relaxed);
-        worker.steal_count.store(self.steal_count, Relaxed);
-        worker
-            .steal_operations
-            .store(self.steal_operations, Relaxed);
-        worker.poll_count.store(self.poll_count, Relaxed);
-
+    pub(crate) fn submit(&mut self, worker: &WorkerMetrics, _mean_poll_time: u64) {
+        #[cfg(tokio_unstable)]
+        {
+            worker.mean_poll_time.store(_mean_poll_time, Relaxed);
+            worker.park_count.store(self.park_count, Relaxed);
+            worker
+                .park_unpark_count
+                .store(self.park_unpark_count, Relaxed);
+            worker.noop_count.store(self.noop_count, Relaxed);
+            worker.steal_count.store(self.steal_count, Relaxed);
+            worker
+                .steal_operations
+                .store(self.steal_operations, Relaxed);
+            worker.poll_count.store(self.poll_count, Relaxed);
+
+            worker
+                .local_schedule_count
+                .store(self.local_schedule_count, Relaxed);
+            worker.overflow_count.store(self.overflow_count, Relaxed);
+
+            if let Some(poll_timer) = &self.poll_timer {
+                let dst = worker.poll_count_histogram.as_ref().unwrap();
+                poll_timer.poll_counts.submit(dst);
+            }
+        }
         worker
             .busy_duration_total
             .store(self.busy_duration_total, Relaxed);
-
-        worker
-            .local_schedule_count
-            .store(self.local_schedule_count, Relaxed);
-        worker.overflow_count.store(self.overflow_count, Relaxed);
-
-        if let Some(poll_timer) = &self.poll_timer {
-            let dst = worker.poll_count_histogram.as_ref().unwrap();
-            poll_timer.poll_counts.submit(dst);
-        }
     }
 
     /// The worker is about to park.
     pub(crate) fn about_to_park(&mut self) {
-        self.park_count += 1;
-        self.park_unpark_count += 1;
-
-        if self.poll_count_on_last_park == self.poll_count {
-            self.noop_count += 1;
-        } else {
-            self.poll_count_on_last_park = self.poll_count;
+        #[cfg(tokio_unstable)]
+        {
+            self.park_count += 1;
+            self.park_unpark_count += 1;
+
+            if self.poll_count_on_last_park == self.poll_count {
+                self.noop_count += 1;
+            } else {
+                self.poll_count_on_last_park = self.poll_count;
+            }
         }
     }
 
     /// The worker was unparked.
     pub(crate) fn unparked(&mut self) {
-        self.park_unpark_count += 1;
+        #[cfg(tokio_unstable)]
+        {
+            self.park_unpark_count += 1;
+        }
     }
```

Comment:
In the rest of the Tokio code base, we do this a different way, so let's stick to that convension. Instead of gating functionality within a function, we have a separate empty function definition when the cfg flag isn't enabled. So this function would become:

```rust
cfg_unstable_metrics! {
    /// The worker was unparked.
    pub(crate) fn unparked(&mut self) {
        self.park_unpark_count += 1;
    }
}

cfg_not_unstable_metrics! {
    /// The worker was unparked.
    pub(crate) fn unparked(&mut self) {}
}
```

Please do the same here. Keep a single `cfg_unstable_metrics` block (and a single `cfg_not_unstable_metrics` block) for all the functions that require this behavior, so that they're grouped together.

For the more complex functions above that have a mix of stablized and unstablized implementation, split the unstablized part out into a separate function with an impl in each of the macro blocks (see example in the comment on `submit`).